hibernate-core/src/main/java/org/hibernate/engine/jdbc/env/internal/ExtractedDatabaseMetaDataImpl.java

```diff
@@ -181,7 +181,8 @@ public Builder apply(DatabaseMetaData databaseMetaData) throws SQLException {
 			connectionCatalogName = databaseMetaData.getConnection().getCatalog();
 			// NOTE : databaseMetaData.getConnection().getSchema() would require java 1.7 as baseline
 			supportsRefCursors = StandardRefCursorSupport.supportsRefCursors( databaseMetaData );
-			supportsNamedParameters = databaseMetaData.supportsNamedParameters();
+			supportsNamedParameters = jdbcEnvironment.getDialect()
+					.supportsNamedParameters() && databaseMetaData.supportsNamedParameters();
```

Comment:
I think the convention for `ExtractedDatabaseMetaDataImpl.Builder` is to collect the unprocessed settings as read from `java.sql.DatabaseMetaData`.

It is in JdbcEnvironmentImpl that overrides take place. Actually, the example I gave of JdbcEnvironmentImpl.nameQualifierSupport is not a good one.

A better example is: https://github.com/hibernate/hibernate-orm/blob/master/hibernate-core/src/main/java/org/hibernate/engine/jdbc/env/internal/JdbcEnvironmentImpl.java#L226

In `JdbcEnvironmentImpl`: 

```
this.extractedMetaDataSupport = new ExtractedDatabaseMetaDataImpl.Builder( this )
		.apply( databaseMetaData )
		.setConnectionSchemaName( determineCurrentSchemaName( databaseMetaData, serviceRegistry, dialect ) )
		.build();
```

could be:

```
this.extractedMetaDataSupport = new ExtractedDatabaseMetaDataImpl.Builder( this )
		.apply( databaseMetaData )
		.setConnectionSchemaName( determineCurrentSchemaName( databaseMetaData, serviceRegistry, dialect ) )
		.setSupportsNamedParameters( dialect.supportsNamedParameters( databaseMetaData ) )
		.build();
```
By default ``Dialect.supportsNamedParameters(DatabaseMetaData dbMetaData)`` can return ``dbMetaData.supportsNamedParameters()``; for ``HSQLDialect``,  false can be returned. 

This way Dialect is free to override, or not.